modules/psap-special-resource-operator-using-CRD.adoc

```diff
@@ -0,0 +1,480 @@
+            spec:
+                remediationAction: enforce
+                severity: low
+                namespaceselector:
+                    exclude:
+                        - kube-*
+                    include:
+                        - '*'
+                object-templates:
+                    - complianceType: musthave
+                      objectDefinition:
+                            apiVersion: cluster.open-cluster-management.io/v1alpha1
+                            kind: ClusterClaim
+                            metadata:
+                                name: osimage.openshift.io
+                            spec:
+                                value: '{{ (lookup "config.openshift.io/v1" "ClusterVersion" "" "version").status.desired.image }}' <1>
+---
+apiVersion: apps.open-cluster-management.io/v1
+kind: PlacementRule
+metadata:
+    name: version-claim-placementrules
+    namespace: default
+spec:
+    clusterSelector:
+        matchExpressions:
+            - key: name
+              operator: NotIn
+              values:
+                - "local-cluster"
+---
+apiVersion: policy.open-cluster-management.io/v1
+kind: PlacementBinding
+metadata:
+    name: version-claim-placementbinding
+    namespace: default
+placementRef:
+    name: version-claim-placementrules
+    kind: PlacementRule
+    apiGroup: apps.open-cluster-management.io
+subjects:
+    - name: version-claim-policy
+      kind: Policy
+      apiGroup: policy.open-cluster-management.io
+----
+<1> This ACM Policy creates the necessary cluster claim and keeps it in sync with changes if the spoke’s OCP version is upgraded.
++
+** You applied the above YAML with the command:
++
+[source,terminal]
+----
+oc apply -f <file name>
+----
+
+.Procedure
+
+. Create a templates directory and change into it.
++
+[source,terminal]
+----
+$ mkdir -p charts/acm-simple-kmod-0.0.1/templates
+----
++
+[source,terminal]
+----
+$ cd charts/acm-simple-kmod-0.0.1/templates
+----
+. Create the buildconfig, policy, and placement.
+.. Save this YAML template for the image stream and build config in the `templates` directory as `0001-buildconfig.yaml`.
++
+[source,yaml]
+----
+apiVersion: build.openshift.io/v1
+kind: BuildConfig
+metadata:
+    labels:
+        app: {{ printf "%s-%s" .Values.specialResourceModule.metadata.name .Values.kernelFullVersion | replace "." "-" | replace "_" "-" | trunc 63 }}
+    name: {{ printf "%s-%s" .Values.specialResourceModule.metadata.name .Values.kernelFullVersion | replace "." "-" | replace "_" "-" | trunc 63 }}
+    annotations:
+specialresource.openshift.io/wait: "true"
+spec:
+    nodeSelector:
+        node-role.kubernetes.io/worker: ""
+    runPolicy: "Serial"
+    triggers:
+        - type: "ConfigChange"
+        - type: "ImageChange"
+    source:
+        git:
+            ref: {{.Values.git.ref}}
+            uri: {{.Values.git.uri}}
+        type: Git
+    strategy:
+        dockerStrategy:
+            dockerfilePath: Dockerfile.SRO
```

Comment:
This changed quite a bit. KVC support has been removed from the simple-kmod repo, so now we need these changes in the manifests: https://github.com/openshift/special-resource-operator/pull/229

Reply:
replaced the text per https://github.com/openshift/special-resource-operator/pull/229/files